src/main/java/core/basesyntax/converter/StringToFruitValidator.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax.converter;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Objects;
+
+public class StringToFruitValidator {
+
+    public boolean validateInputData(String[] data) {
+        return isDataValid(data)
+                && isOperatorValid(data)
+                && isNameValid(data)
+                && isQuantityValid(data);
+    }
+
+    private boolean isDataValid(String[] data) {
+        return data.length == 3;
+    }
+
+    private boolean isOperatorValid(String[] data) {
+        try {
+            FruitTransaction.Operation.fromCode(data[0]);
+            return true;
+        } catch (RuntimeException e) {
+            throw new RuntimeException("Invalid operator: " + data[0]);
```

Comment:
The `isOperatorValid` method throws a `RuntimeException` with a generic message. Consider using a more specific exception type, such as `IllegalArgumentException`, and providing a more detailed message to help with debugging.